repos/system_upgrade/el7toel8/actors/kernel/checkkernelmodules/actor.py

```diff
@@ -0,0 +1,52 @@
+from leapp.actors import Actor
+from leapp.libraries.common.reporting import report_with_remediation
+from leapp.models import ActiveKernelModulesFacts, WhitelistedKernelModules
+from leapp.reporting import Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+class CheckKernelModules(Actor):
+    """
+    Actor checks if any loaded RHEL 7 kernel module is missing in the RHEL 8.
+    If yes, the upgrade process will be inhibited.
+
+    Inhibition is done because missing kernel module on the RHEL 8 system may
+    mean that the hardware using such module as a driver would not work on the
+    RHEL 8.
+
+    Note:
+     - List of kernel modules missing on the RHEL 8 system is located in the
+        files/removed_modules.txt file.
+    """
+    name = "check_kernel_modules"
+    consumes = (ActiveKernelModulesFacts, WhitelistedKernelModules)
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        with open("files/removed_modules.txt", "r") as removed:
+            removed_modules = [x.strip() for x in removed.readlines()]
+            whitelisted_modules = set()
+            modules_to_report = set()
+
+            for fact in self.consume(WhitelistedKernelModules):
+                whitelisted_modules.update(fact.whitelisted_modules)
+
+            for fact in self.consume(ActiveKernelModulesFacts):
+                for active_module in fact.kernel_modules:
+                    if active_module.filename in whitelisted_modules:
+                        continue
+                    if active_module.filename in removed_modules:
+                        modules_to_report.add(active_module.filename)
```

Comment:
Before inhibiting, check if the driver is used by any device: `udevadm info -e` and look for `DRIVER=`.